src/vm.rb

```diff
@@ -61,6 +61,61 @@ def destroy
 
     private
 
+    CERTS_DIR = '/var/lib/metalware/certs'
+
+    def libvirt_host
+      node.config.libvirt_host
+    end
+
+    def libvirt
+      Libvirt.open("qemu://#{libvirt_host}/system")
+    end
+
+    def certificate?
+      File.exist?("#{CERTS_DIR}/#{libvirt_host}-key.pem")
```

Comment:
This file path is duplicated here and in `generate_certificate_key` and `generate_server_certificates` below, so it would make sense to split out a `certificate_key_path` method for it to remove this duplication.

While doing that it may be worthwhile to use `File.join` rather than string interpolation to form this file path - I'm not sure if it will make much difference in this case but generally this is a good practice as it:

a. makes the code less dependent on a particular platform separator (in this case `/`);
b. means you'll still get a nice file path if the things to be joined themselves start or end with a separator;
c. makes your intention obvious.

Reply:
Should I be using `File.join` everywhere I'm defining file paths?

Reply:
Ah, you've said yes later on.. ignore!